src/slider/__tests__/slider.test.js

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import React from 'react';
+import {mount} from 'enzyme';
+import {Slider, StyledThumb} from '../index';
+
+/* global document */
+/* global window */
+
+const originalDocumentAddListener = document.addEventListener;
+
+describe('Stateless slider', function() {
+  let wrapper;
+  let allProps: {
+      range: Array<number>,
+      value: Array<number>,
+      step: ?number,
+      onChange: () => void,
+    },
+    mockFn;
+
+  beforeEach(function() {
+    //$FlowFixMe
+    document.addEventListener = jest.fn();
+    mockFn = jest.fn();
+    allProps = {
+      onChange: mockFn,
+      range: [0, 100],
+      value: [60],
+      step: 0,
+    };
+  });
+
+  afterEach(function() {
+    jest.restoreAllMocks();
+    document.addEventListener.mockClear();
+    wrapper && wrapper.unmount();
+    //$FlowFixMe
+    document.addEventListener = originalDocumentAddListener;
```

Comment:
We don't need to restore it after every single test but can do it in the `afterAll` instead

Reply:
restore it in `afterAll` is not safe to preserve isolation of unit-test and may cause conflicts in environments. We already have some code in other places, which does similar

Reply:
ok, moved to afterAll